src/main/bx/models/loaders/IDocumentLoader.bx

```diff
@@ -0,0 +1,118 @@
+/**
+ * [BoxLang]
```

Comment:
Add progress tracking

/**
 * Set progress callback for loading
 *
 * @callback Function: ( completed, total, currentDoc ) => {}
 *
 * @return IDocumentLoader for chaining
 */
IDocumentLoader function onProgress( required function callback );

aiDocument( "/large-dataset" )
    .onProgress( ( completed, total, doc ) => {
        println( "Loading: #completed#/#total# - #doc.metadata.source#" );
    } )
    .load();